KomaMRICore/test/runtests.jl

```diff
@@ -458,3 +419,89 @@ end
     @suppress print_devices()
     @test true
 end
+
+# --------- Motion-related tests -------------
+# We compare with the result given by OrdinaryDiffEqTsit5
+@testitem "Motion" tags=[:core, :motion] begin 
+    using Suppressor, OrdinaryDiffEqTsit5
+    include("initialize_backend.jl")
+
+    Nadc = 25
+    M0 = 1.0
+    T1 = 100e-3
+    T2 = 10e-3
+    B1 = 20e-6
+    Trf = 3e-3
+    γ = 2π * 42.58e6
+    φ = π / 4
+    B1e(t) = B1 * (0 <= t <= Trf)
+    duration = 2*Trf
+
+    Gx = 1e-3
+    Gy = 1e-3
+    Gz = 0
+
+    motions = [
+        Translate(0.0, 0.1, 0.0, TimeRange(0.0, 1.0)),
+        Rotate(0.0, 0.0, 45.0, TimeRange(0.0, 1.0)),
+        HeartBeat(-0.6, 0.0, 0.0, Periodic(1.0)),
+        Path([0.0 0.0], [0.0 1.0], [0.0 0.0], TimeRange(0.0, 10.0)),
+        FlowPath([0.0 0.0], [0.0 1.0], [0.0 0.0], [0.0 0.0], TimeRange(0.0, 10.0))
+    ]
+
+    x0 = [0.1]
+    y0 = [0.1]
+    z0 = [0.0]
+
+    for m in motions
+        motion = MotionList(m)
+
+        coords(t) = get_spin_coords(motion, x0, y0, z0, t)
+        x(t) = (coords(t)[1])[1]
+        y(t) = (coords(t)[2])[1]
+        z(t) = (coords(t)[3])[1]
+
+        ## Solving using DiffEquations.jl
+        function bloch!(dm, m, p, t)
+            mx, my, mz = m
+            bx, by, bz = [B1e(t) * cos(φ), B1e(t) * sin(φ), (x(t) * Gx + y(t) * Gy + z(t) * Gz)]
+            dm[1] = -mx / T2 + γ * bz * my - γ * by * mz
+            dm[2] = -γ * bz * mx - my / T2 + γ * bx * mz
+            dm[3] =  γ * by * mx - γ * bx * my - mz / T1 + M0 / T1
+            return nothing
+        end
+        m0 = [0.0, 0.0, 1.0]
+        tspan = (0.0, duration)
+        prob = ODEProblem(bloch!, m0, tspan)
+        # Only at ADC points
+        tadc = range(Trf, duration, Nadc)
+        sol = @time solve(prob, Tsit5(), saveat = tadc, abstol = 1e-9, reltol = 1e-9)
+        sol_diffeq = hcat(sol.u...)'
+        mxy_diffeq = sol_diffeq[:, 1] + im * sol_diffeq[:, 2]
+
+        ## Solving using KomaMRICore.jl
+        # Creating Sequence
+        seq = Sequence()
+        seq += RF(cis(φ) .* B1, Trf)
+        seq.GR[1,1] = Grad(Gx, duration)
+        seq.GR[2,1] = Grad(Gy, duration)
+        seq.GR[3,1] = Grad(Gz, duration)
+        seq.ADC[1] = ADC(Nadc, duration-Trf, Trf)
+        # Creating object
+        obj = Phantom(x = x0, y = y0, z = z0, ρ = [M0], T1 = [T1], T2 = [T2], motion = motion)
+        # Scanner
+        sys = Scanner()
+        # Simulation
+        for sim_method in [KomaMRICore.Bloch(), KomaMRICore.BlochSimple(), KomaMRICore.BlochDict()]
+            sim_params = Dict{String, Any}(
+                "sim_method"=>sim_method,
+                "return_type"=>"mat"
+            )
+            raw_aux = @suppress simulate(obj, seq, sys; sim_params)
+            raw = raw_aux[:, 1, 1]
```

Comment:
Add GPU variable in sim_params.
```suggestion
        for sim_method in [KomaMRICore.Bloch(), KomaMRICore.BlochSimple(), KomaMRICore.BlochDict()]
            sim_params = Dict{String, Any}(
                "sim_method"=>sim_method,
                "return_type"=>"mat"
            )
            raw = @suppress simulate(obj, seq, sys; sim_params)[:, 1, 1]
```
Also, it would not be considered if I create a new simulation method.